cibyl/utils/query_cache.py

```diff
@@ -0,0 +1,57 @@
+"""
+#    Copyright 2022 Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+
+
+class QueriesCache:
```

Comment:
I think this patch is a great idea and a must for Cibyl, the only thing I'm not sure about is actually the name. Is it really considered a cache? I was always considering a cache as something for storing data to make future queries/requsts much quicker.

In the future, I think Cibyl will have a cache, where we will save data we pulled, either locally or remotely, but that's not what this PR implements. So from my side, if you think the name fits and there is no better naming for this mechanism, you can go ahead and merge

Reply:
That's a good point, naming is hard. I'll think of a less potentially ambiguous name